ee/server/service/mdm.go

```diff
@@ -111,3 +112,43 @@ func (svc *Service) MDMAppleEraseDevice(ctx context.Context, hostID uint) error
 	}
 	return nil
 }
+
+// TODO: this probably shouldn't be a public service method that deals with
+// authorization logic as is not directly called from an endpoint handler. Some
+// ideas:
+//
+// - use this logic in the cron job that installs configuration profiles
+// - use this logic in the service methods that modify app/team configs and
+// call it if filevault is turned on/off
```

Comment:
@mna do you have any thoughts about this? this method builds the configuration profile and then does an `INSERT` into `mdm_apple_configuration_profiles` so the cron job can pick it and install it in the hosts that need the profile.

The question now is: where should this logic live? The first bullet is very appealing, the only drawback is that the host won't be `'pending'` instantaneously, but maybe that can be solved as part of https://github.com/fleetdm/fleet/issues/10122?

Alternatively, we can do it as part of the logic you're building.

Reply:
Hey @roperzh , that looks great! Agree that authz should probably not be done here as it's meant to be called from elsewhere/already authz places. I think it makes sense for it to be called from the logic I'm building - when enabling for a team/no team, call this to generate the config profiles (but as you say, _not_ to actually deploy to all impacted hosts, leaving that to the cron job as it would make the endpoint call way too slow and not robust).

Reply:
I can certainly push my PR without that call and add it later, to minimize dependencies between our work.

Reply:
@mna thank you! sounds good! Let's do that! additionally we need another method to _remove_ the profile if FileVault is turned off, I can add that method as part of this PR too if that sounds good.